saleor/graphql/product/resolvers.py

```diff
@@ -65,3 +65,7 @@ def resolve_products(info, category_id, query):
 
 def resolve_product_types():
     return models.ProductType.objects.all().distinct()
+
+
+def resolve_product_variants(info, ids):
```

Comment:
To unify this endpoint the rest of connection-endpoints in our API I would make the `ids` argument optional. If it's not provided the endpoint would behave just as all other paginated connections. But when providing the list of `ids` we would filter the results and only return the requested IDs.

Reply:
That requires some extra permissions. A normal user won't be able to list all checkouts only its own.

Reply:
I assume you'll use a different query for checkouts, this one only returns variants. But it is true that similarly to how we filter products in the `products` query, here we should make sure that we don't expose variants of unpublished products publicly.

Reply:
Sorry for the misunderstanding, mu ming was attached to checkout API.

Reply:
Wat? 😄

Reply:
![image](https://user-images.githubusercontent.com/5421321/45087612-d1a56e00-b106-11e8-8991-bf5addac43f7.png)
(credits: @patrys)